src/server/scripts/Kalimdor/zone_silithus.cpp

```diff
@@ -1328,61 +1379,124 @@ class go_wind_stone : public GameObjectScript
                             break;
                         case GOSSIP_ACTION_INFO_DEF + 9:
-                            SummonNPC(me, player, NPC_DUKE_EARTH, SPELL_DUKE_EARTH);
+                            if (rank >= 2)
+                            {
+                                SummonNPC(me, player, NPC_DUKE_EARTH, SPELL_DUKE_EARTH);
+                            }
                             break;
                         case GOSSIP_ACTION_INFO_DEF + 10:
-                            SummonNPC(me, player, NPC_DUKE_AIR, SPELL_DUKE_AIR);
+                            if (rank >= 2)
+                            {
+                                SummonNPC(me, player, NPC_DUKE_AIR, SPELL_DUKE_AIR);
+                            }
                             break;
 
                         case GOSSIP_ACTION_INFO_DEF + 11:
-                            SummonNPC(me, player, RAND(NPC_ROYAL_FIRE, NPC_ROYAL_AIR, NPC_ROYAL_EARTH, NPC_ROYAL_WATER), SPELL_ROYAL_RANDOM);
+                            if (rank == 3)
+                            {
+                                SummonNPC(me, player, RAND(NPC_ROYAL_FIRE, NPC_ROYAL_AIR, NPC_ROYAL_EARTH, NPC_ROYAL_WATER), SPELL_ROYAL_RANDOM);
+                            }
                             break;
                         case GOSSIP_ACTION_INFO_DEF + 12:
-                            SummonNPC(me, player, NPC_ROYAL_FIRE, SPELL_ROYAL_FIRE);
+                            if (rank == 3)
+                            {
+                                SummonNPC(me, player, NPC_ROYAL_FIRE, SPELL_ROYAL_FIRE);
+                            }
                             break;
                         case GOSSIP_ACTION_INFO_DEF + 13:
-                            SummonNPC(me, player, NPC_ROYAL_WATER, SPELL_ROYAL_WATER);
+                            if (rank == 3)
+                            {
+                                SummonNPC(me, player, NPC_ROYAL_WATER, SPELL_ROYAL_WATER);
+                            }
                             break;
                         case GOSSIP_ACTION_INFO_DEF + 14:
-                            SummonNPC(me, player, NPC_ROYAL_EARTH, SPELL_ROYAL_EARTH);
+                            if (rank == 3)
+                            {
+                                SummonNPC(me, player, NPC_ROYAL_EARTH, SPELL_ROYAL_EARTH);
+                            }
                             break;
                         case GOSSIP_ACTION_INFO_DEF + 15:
-                            SummonNPC(me, player, NPC_ROYAL_AIR, SPELL_ROYAL_AIR);
+                            if (rank == 3)
+                            {
+                                SummonNPC(me, player, NPC_ROYAL_AIR, SPELL_ROYAL_AIR);
+                            }
                             break;
 
                         default:
                             break;
                     }
                     return true;
                 }
+                void Reset() override
```

Comment:
```suggestion

                void Reset() override
```
Adding an empty line between the 2 functions